### PR TITLE
feat: bump to 0.3 alpha

### PR DESCRIPTION
**Pull request by driftluo**
- Upgrade moleculec to 0.5.0
- Upgrade to async ecology

### CODE REVIEW

File: protocols/discovery/src/lib.rs [hunk 1]
```diff
@@ -163,7 +163,7 @@ pub struct Discovery<M> {
 
     dynamic_query_cycle: Option<Duration>,
 
-    check_interval: Interval,
+    check_interval: Option<Interval>,
```
Comment (doitian) [inline protocols/discovery/src/lib.rs]: Why this is changed to an option?

File: src/service/config.rs [hunk 2]
```diff
@@ -33,6 +36,39 @@ impl Default for ServiceConfig {
     }
 }
 
+#[derive(Clone, Copy)]
```
Comment (doitian) [inline src/service/config.rs]: This looks like a refactoring, it's better in its own PR.

File: protocols/discovery/src/lib.rs (hunk 1 above)
Comment (driftluo) [reply protocols/discovery/src/lib.rs]: tokio's time now depends on the runtime global handle and cannot be created outside the runtime

File: src/service/config.rs (hunk 2 above)
Comment (driftluo) [reply src/service/config.rs]: This involves the external interface of `yamux`, which must be merged before the release.

Comment (driftluo) [reply src/service/config.rs]: https://github.com/nervosnetwork/p2p/pull/209

**Pull request merged**